### PR TITLE
MultipleItemEvent

### LINKED ISSUE #950: InventoryPreClickEvent and PlayerSwapItemEvent don't extend ItemEvent
ItemEvent only provides one item.

But InventoryPreClickEvent and PlayerSwapItemEvent needs 2 Items so they don't extend ItemEvent. Understand.

But maybe add an MultiItemEvent or so with an List<ItemStack> and make it work with EventFilter.ITEM

### PR DESCRIPTION
**Pull request by DasLixou**
This adds support for `EventFilter.ITEM` for Events with more than one ItemStack.

before, even:

```java
EventNode.event("multipleItemEventTestlol", EventFilter.ITEM, itemEvent -> {
        return test(itemEvent.getItemStack());
});
```
wasn't possible when using multiple item events like PlayerSwapItemEvent.

now, with this Pull Request, this is possible:

```java
EventNode.event("multipleItemEventTestlol", EventFilter.ITEM, itemEvent -> {
        if(!(itemEvent instanceof MultipleItemEvent)) return test(itemEvent.getItemStack());
        boolean valid = false;
        for (ItemStack item : ((MultipleItemEvent)itemEvent).getItemStacks()) {
                if(test(item)) {
                        valid = true;
                        break;
                }
        }
        return valid;
});
```

This fixes #950 

Feedback welcome - Thanks 😃

### CODE REVIEW

File: src/main/java/net/minestom/server/event/inventory/InventoryClickEvent.java [hunk 1]
```diff
@@ -80,11 +79,14 @@ public ItemStack getClickedItem() {
...
     public ItemStack getCursorItem() {
-        return cursorItem;
+        return itemStacks[1];
     }
 
     @Override
     public @Nullable Inventory getInventory() {
         return inventory;
     }
+
```
Comment (WeiiswurstDev) [inline src/main/java/net/minestom/server/event/inventory/InventoryClickEvent.java:89]: Consider adding some javadoc here (size of the array, indexes of the contents)

File: src/main/java/net/minestom/server/event/inventory/InventoryItemChangeEvent.java [hunk 2]
```diff
@@ -54,11 +53,14 @@ public int getSlot() {
...
     public @NotNull ItemStack getNewItem() {
-        return newItem;
+        return itemStacks[1];
     }
 
     @Override
     public @Nullable Inventory getInventory() {
         return inventory;
     }
+
```
Comment (WeiiswurstDev) [inline src/main/java/net/minestom/server/event/inventory/InventoryItemChangeEvent.java:63]: Consider adding some javadoc here (size of the array, indexes of the contents)

File: src/main/java/net/minestom/server/event/inventory/InventoryClickEvent.java (hunk 1 above)
Comment (WeiiswurstDev) [reply src/main/java/net/minestom/server/event/inventory/InventoryClickEvent.java:89]: Ok for some reason the preview does not show the right lines lol

**Comment by WeiiswurstDev** [comment]
Generally getItemStacks() needs javadoc wherever it is implemented because it is unclear what is in that array

File: src/main/java/net/minestom/server/event/inventory/InventoryItemChangeEvent.java (hunk 2 above)
Comment (DasLixou) [reply src/main/java/net/minestom/server/event/inventory/InventoryItemChangeEvent.java:63]: Sure I’ll do that tomorrow. Thanks for the feedback 😄

**Comment by DasLixou** [comment]
> Generally getItemStacks() needs javadoc wherever it is implemented because it is unclear what is in that array

Yeah I’ll make javadoc for them tomorrow :)

**Comment by TheMode** [comment]
I am personally against the change. It does not seem to add much except confusion (also, arrays are very unsafe to expose like that since you cannot guarantee immutability)

**Comment by WeiiswurstDev** [comment]
I think it would work better as a 'tag' interface of sorts. 

Since you have different values in the array every time anyways, they cant really be used without checking the type of event first. And if the type of the event has to be checked anyways, then you can just use the existing methods of the events.

tldr the only reason to use it is to cancel events with the filter, and we dont need to expose an additional method for that purpose.
